hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/DefaultSource.scala

```diff
@@ -295,8 +295,11 @@ object DefaultSource {
       }
 
       lazy val enableFileGroupReader = SparkConfigUtils
-        .getStringWithAltKeys(parameters, HoodieReaderConfig.FILE_GROUP_READER_ENABLED).toBoolean &&
-        !metaClient.isMetadataTable
+        .getStringWithAltKeys(parameters, HoodieReaderConfig.FILE_GROUP_READER_ENABLED).toBoolean
+      if (!enableFileGroupReader) {
+        throw new IllegalArgumentException("File group reader is disabled")
+      }
```

Comment:
Should the `enableFileGroupReader` be removed if it is not allowed to use non-fg reader code path on the read side?